components/omega/doc/design/OmegaV1GoverningEqns.md

```diff
@@ -0,0 +1,862 @@
+
+For any three-dimensional quantity $\varphi(x,y,z,t)$, the mass-thickness-averaged quantity in layer $k$ is defined as
+
+$$
+\varphi_k(x,y,t)
+\equiv \frac{\int_{z_k^{bot}}^{z_k^{top}} \rho \varphi dz}{\int_{z_k^{bot}}^{z_k^{top}} \rho dz}
+= \frac{\int_{z_k^{bot}}^{z_k^{top}} \rho \varphi dz}{h_k}
+$$(def-mass-thickness-average)
+
+At this point our derivation has not made any assumptions about density, and may be used for both Boussinesq and non-Boussinesq fluids. A Boussinesq derivation would now assume small variations in density and replace $\rho(x,y,z,t)$ with a constant $\rho_0$ everywhere but the pressure gradient coefficient. In that case $\rho$ divides out in [](#def-mass-thickness-average) the Boussinesq layer quantities would simply be thickness-weighted averages.
+
+We can now derive the layered equations. Integrate the continuous equations [](continuous-momentum-final), [](continuous-mass-final), [](continuous-tracer-final) in $z$ from $z_k^{bot}$ to $z_k^{top}$,
+
+$$
+\frac{\partial }{\partial t} \int_{z_k^{bot}}^{z_k^{top}} \rho {\bf u} dz
++  \int_{z_k^{bot}}^{z_k^{top}} \nabla \cdot (\rho \mathbf{u} \otimes \mathbf{u}) dz
++  \int_{z_k^{bot}}^{z_k^{top}} \frac{\partial }{\partial z} \left(  \rho w{\bf u} \right) dz
++  \int_{z_k^{bot}}^{z_k^{top}} 
+ f \rho {\bf u}^\perp dz
+  =  \int_{z_k^{bot}}^{z_k^{top}}  \left[- \nabla_z p 
+  - \rho \nabla_z \Phi 
++ \rho {\bf D}^u + \rho {\bf F}^u \right] dz
+$$ (z-integration-momentum)
+
+$$
+\frac{\partial }{\partial t}  \int_{z_k^{bot}}^{z_k^{top}} \rho dz
+ +\nabla_z \cdot \left( \int_{z_k^{bot}}^{z_k^{top}} \rho  {\bf u} dz \right)
+ + \int_{z_k^{bot}}^{z_k^{top}}\frac{\partial}{\partial z} \left(  \rho  w \right) dz
+= 0
+$$ (z-integration-mass)
+
+$$
+\frac{\partial }{\partial t} \int_{z_k^{bot}}^{z_k^{top}} \rho \varphi  dz
+ + \nabla_z \cdot \left( \int_{z_k^{bot}}^{z_k^{top}} \rho \varphi {\bf u} dz  \right)
+ + \int_{z_k^{bot}}^{z_k^{top}} \frac{\partial}{\partial z} \left(  \rho \varphi w \right) dz
+= \int_{z_k^{bot}}^{z_k^{top}} \left( D^\varphi + Q^\varphi \right) dz
+$$ (z-integration-tracers)
+ 
+This results in conservation equations that are valid over the layer. The momentum variables are simply vertically averaged. Tracer variables are vertically mass-averaged, as defined in [](def-mass-thickness-average).
+
+In order to deal with nonlinear terms where we take the integrals of products, we may assume that the variables are piecewise constant in the vertical within each layer, i.e.
+
+$$
+a(x,y,z,t) = a_k(x,y,t) \in [z_k^{bot}, z_k^{top}).
+$$ (discrete-a)
+
+Then variables may come out of the integral as needed. For example,
```

Comment:
I continue to think all integrals really *must* be in $p$ space (weighted by $\rho$), rather than in $z$.